.compute

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+# this script gets executed on every node of an allocation;
+# in our case we use the provided COMPUTE_* env variables
+# to construct a distributed TensorFlow cluster definition
+
+set -e
+
+# activating standard TensorFlow Python virtual environment
+source /usr/local/tensorflow/bin/activate
+
+# the standard script to execute
+base_cmd="bin/run-wer-automation.sh"
+
+# reading the comma separated node list into array "nodes"
+IFS=',' read -r -a nodes <<< "$COMPUTE_NODES"
+# keep fist node for convenience
+first=${nodes[0]}
+# hostname for debugging
+hostname=`hostname`
+
+if ((${#nodes[@]} == 1)); then
+    # there is only one (this) node - so we are alone and we don't need a cluster definition
+    echo "[slurm   ] Starting single node process on $hostname ..."
+    $base_cmd --job_name localhost "$@"
+else
+    # there is more than on node so we will build a cluster definition
+
+    # defining all cluster ports in a way that avoids collisions with other cluster allocations
+    # (that could eventually get scheduled on the same node)
+    ((port_base=10000 + (COMPUTE_JOB_NUMBER * 100) % 50000))
```

Comment:
It's because of 16 bit port numbers and a bit laziness.
Lower bound 10000 due to the habit of many devs to open their application ports somewhere between 1024 and 9999.
Upper bound 60000 because its the last 10000 dividable before 16 bit maximum.
Modulo 100 window due to a rough estimate of our future cluster sizes - I felt 10 being too low and 1000 exaggerated.

You can provide a better formula, if you want. This is just my first draft/estimate.